### PR TITLE
Add log if connecting to redis cluster failed

### PR DESCRIPTION
**Pull request by hamdanjaveed**

### CODE REVIEW

File: misk-redis/src/main/kotlin/misk/redis/RedisModule.kt [hunk 1]
```diff
@@ -35,10 +37,19 @@ class RedisModule(
...
     // Authenticate with the redis server
-    jedisPool.resource.use {
-      it.auth(replicationGroup.redis_auth_password)
+    try {
+      jedisPool.resource.use { it.auth(replicationGroup.redis_auth_password) }
+    } catch(e: JedisConnectionException) {
+      log.error { "Failed to connect to redis cluster: " +
+        "${replicationGroup.writer_endpoint.hostname}:${replicationGroup.writer_endpoint.port}" }
+
+      e.printStackTrace()
```
Comment (ryanhall07) [inline misk-redis/src/main/kotlin/misk/redis/RedisModule.kt]: don't print the stack trace to the console, just log the exception above

File: misk-redis/src/main/kotlin/misk/redis/RedisModule.kt [hunk 2]
```diff
@@ -35,10 +37,19 @@ class RedisModule(
     )
 
     // Authenticate with the redis server
-    jedisPool.resource.use {
-      it.auth(replicationGroup.redis_auth_password)
+    try {
```
Comment (ryanhall07) [inline misk-redis/src/main/kotlin/misk/redis/RedisModule.kt]: this behavior is different now since you swallowed the exception and did not propagate

Comment (hamdanjaveed) [reply misk-redis/src/main/kotlin/misk/redis/RedisModule.kt]: Re-throwing the exception now

File: misk-redis/src/main/kotlin/misk/redis/RedisModule.kt [hunk 3]
```diff
@@ -35,10 +37,19 @@ class RedisModule(
...
+      jedisPool.resource.use { it.auth(replicationGroup.redis_auth_password) }
+    } catch(e: JedisConnectionException) {
+      log.error { "Failed to connect to redis cluster: " +
```
Comment (ryanhall07) [inline misk-redis/src/main/kotlin/misk/redis/RedisModule.kt]: an alternative approach is to throw a new exception that holds the context you need. that way you don't have a separate error message that is unconnected to the final stack trace.

```
class RedisConnectionException : RuntimeException(ReplicationGroup, Cause) {
  super("Failed to connect...", cause);
}

catch (e: JedisConnectionException) {
  throw RedisConnectionException(replicationGroup, e);
}
```

Comment (hamdanjaveed) [reply misk-redis/src/main/kotlin/misk/redis/RedisModule.kt]: Nice I like that, I'll switch to creating a new exception

File: misk-redis/src/main/kotlin/misk/redis/RedisModule.kt [hunk 4]
```diff
@@ -35,10 +37,17 @@ class RedisModule(
...
+      jedisPool.resource.use { it.auth(replicationGroup.redis_auth_password) }
+    } catch(e: JedisConnectionException) {
+      throw RedisConnectionException(
+        "${replicationGroup.writer_endpoint.hostname}:${replicationGroup.writer_endpoint.port}", e)
     }
 
     return RealRedis(jedisPool)
   }
+
+  companion object {
```
Comment (ryanhall07) [inline misk-redis/src/main/kotlin/misk/redis/RedisModule.kt]: nit: remove the unnecessary log

**Comment by ryanhall07** [comment]
also a general comment, you should refactor the `RedisModule` to be side effect free and not create redis connections from the constructor. I know you have it lazily provided with a Provider, but it would be even better to move the connection logic to a `start` method. This will require a wrapper object, that delays constructing the Redis object until the `start` method is called.

https://github.com/google/guice/wiki/ModulesShouldBeFastAndSideEffectFree

**Pull request merged**